<?xml version="1.0"?>
<launch>
    <arg name="paused" default="false" />
    <arg name="use_sim_time" default="true" />
    <arg name="gui" default="true" />
    <arg name="headless" default="false" />
    <arg name="debug" default="false" />

    <include file="$(find gazebo_ros)/launch/empty_world.launch">
        <arg name="debug" value="$(arg debug)" />
        <arg name="gui" value="$(arg gui)" />
        <arg name="paused" value="$(arg paused)" />
        <arg name="use_sim_time" value="$(arg use_sim_time)" />
        <arg name="headless" value="$(arg headless)" />
    </include>

    <group ns='agv_part'>
        <param name="robot_description"
            command="$(find xacro)/xacro '$(find plane_part)/urdf/agv_moving_part.urdf.xacro'" />

        <node name="agv1_spawner" pkg="gazebo_ros" type="spawn_model"
            respawn="false" output="screen"
            args="-urdf -model agv_moving_part -param robot_description -z 0.05 -x 2.0" />

        <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher"
            respawn="false" output="screen">
            <param name="publish_frequency" type="double" value="50.0" />
            <param name="tf_prefix" type="string" value="agv_part" />
        </node>
    </group>

    <group ns='modulating_part'>
        <param name="robot_description"
            command="$(find xacro)/xacro '$(find plane_part)/urdf/plane_part.urdf.xacro'" />

        <node name="modulating_spawner" pkg="gazebo_ros" type="spawn_model"
            respawn="false" output="screen"
            args="-urdf -model modulating_part -param robot_description" />

        <rosparam file="$(find plane_part)/config/plane_part_controllers.yaml" command="load" />

        <node name="controller_spawner" pkg="controller_manager" type="spawner"
            args="joint_state_controller
                targeting_controller"
            respawn="false" output="screen" />

        <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher"
            respawn="false" output="screen">
            <param name="publish_frequency" type="double" value="50.0" />
            <param name="tf_prefix" type="string" value="modulating_part" />
        </node>

    </group>

    <node
        name="rviz"
        pkg="rviz"
        type="rviz"
        args="-d $(find plane_part)/assemble.rviz" />

</launch>